youtube_dl/extractor/ciscolive.py

```diff
@@ -65,45 +82,49 @@ def _parse_rf_item(self, rf_item):
 
 
 class CiscoLiveSessionIE(CiscoLiveBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?ciscolive(?:\.cisco)?\.com/[^#]*#/session/(?P<id>[^/?&]+)'
+    _VALID_URL = r'https?://(?:www\.)?ciscolive\.com/[^#]*#/session/(?P<id>[^/?&]+)'
     _TESTS = [{
-        'url': 'https://ciscolive.cisco.com/on-demand-library/?#/session/1423353499155001FoSs',
-        'md5': 'c98acf395ed9c9f766941c70f5352e22',
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?search=#/session/16360600004400017rMx',
+        'md5': 'e0f5b0b2927b586ebff619294fec6926',
         'info_dict': {
-            'id': '5803694304001',
+            'id': '6128601216001',
             'ext': 'mp4',
-            'title': '13 Smart Automations to Monitor Your Cisco IOS Network',
-            'description': 'md5:ec4a436019e09a918dec17714803f7cc',
-            'timestamp': 1530305395,
-            'upload_date': '20180629',
+            'title': 'A Deeper Dive into the Telco Cloud Architecture Evolution to support 5G and MEC - BRKSPG-1565',
+            'description': 'md5:04bc54ec8ede2bbd9d21264bfaf16cb3',
+            'timestamp': 1580474594,
+            'upload_date': '20200131',
             'uploader_id': '5647924234001',
-            'location': '16B Mezz.',
         },
     }, {
         'url': 'https://www.ciscolive.com/global/on-demand-library.html?search.event=ciscoliveemea2019#/session/15361595531500013WOU',
         'only_matching': True,
     }, {
         'url': 'https://www.ciscolive.com/global/on-demand-library.html?#/session/1490051371645001kNaS',
         'only_matching': True,
+    }, {
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?#/session/16360600774720017Iby',
+        'only_matching': True,
     }]
 
     def _real_extract(self, url):
         rf_id = self._match_id(url)
         rf_result = self._call_api('session', rf_id, {'id': rf_id}, url)
+        if int(rf_result['responseCode']) != 0:
```

Comment:
Maybe move the error check into `_call_api()` since it's already liable to raise an exception? You could make it so that if the method returns its value is guaranteed to be a `dict`.

Reply:
moved into `_call_api()` as suggested.

if the api call from `_call_api()`  (`_download_json`) succeeds at an http level but the api response produces a **non** 0 (success) error code an ExtractionError is raised.

Only when the ciscolive api(s) return the success code will `_call_api()` return the json dict.

https://github.com/L0wbyte/youtube-dl/commit/9409e00d95c0f0ad656349bb43e449233a66c7c9#diff-39087b9fd88e46ad9ea0b1725fce45470a91a8eecccda1891c129b0794ae6e7fR54-R57